tpls/desul/include/desul/atomics/Compare_Exchange_OpenACC.hpp

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright (c) 2019, Lawrence Livermore National Security, LLC
+and DESUL project contributors. See the COPYRIGHT file for details.
+Source: https://github.com/desul/desul
+
+SPDX-License-Identifier: (BSD-3-Clause)
+*/
+
+#ifndef DESUL_ATOMICS_COMPARE_EXCHANGE_OPENACC_HPP_
+#define DESUL_ATOMICS_COMPARE_EXCHANGE_OPENACC_HPP_
+
+#include <openacc.h>
+
+#include <desul/atomics/Common.hpp>
+#include <desul/atomics/Thread_Fence_OpenACC.hpp>
+#include <type_traits>
+
+namespace desul {
+namespace Impl {
+
+#ifdef __NVCOMPILER
+
+#pragma acc routine seq
+template <class T, class MemoryOrder, class MemoryScope>
+T device_atomic_exchange(T* dest, T value, MemoryOrder, MemoryScope /*scope*/) {
+  if constexpr (std::is_arithmetic_v<T> && ((sizeof(T) == 4) || (sizeof(T) == 8))) {
+    T return_val;
+#pragma acc atomic capture
+    {
+      return_val = *dest;
+      *dest = value;
+    }
+    return return_val;
+  } else {
+    // FIXME_OPENACC
+    printf(
+        "DESUL error in device_atomic_exchange(): Not supported atomic operation in "
+        "the OpenACC backend\n");
+    //  Acquire a lock for the address
+    // while (!lock_address_openacc((void*)dest, scope)) {
+    // }
+    // device_atomic_thread_fence(MemoryOrderAcquire(), scope);
+    T return_val = *dest;
+    *dest = value;
```

Comment:
Why are we not aborting if it's not supported?

Reply:
Same thing forwarding the question to Seyong and not blocking this PR on it.

Reply:
Wasn't the reason that this code path is also used for the host and it works with the `Serial` backend? AFAICT that this only applies for nvc++  and we should be able to abort otherwise.

Reply:
Also no abort on the device with OpenACC
https://github.com/kokkos/kokkos/blob/91ee4e1e145b4daf69c6457ab35c7be0a4cda9f2/core/src/Kokkos_Abort.hpp#L86-L87